### DISCUSSION TITLE
No TSi/TSr sent in IKE_AUTH

### DISCUSSION THREAD

**Discussion by grahamqs** [strongswan/strongswan#1118] [Q&A] [answered]
Hi

I've noticed that traffic selectors are not sent in IKE_AUTH. I have the following configuration in swanctl;

connections {
   home {
      remote_addrs = xxxx
      vips = 0.0.0.0
      mobike = no
      local {
         auth = pubkey
         certs = moonCert1.pem
         id = moon1.test.com
      }
      remote {
         auth = pubkey
      }
      children {
         home {
            remote_ts = 100.0.0.0/8
            esp_proposals = aes128-sha256-ecp256
         }
      }
      version = 2
      proposals = aes128-aes256-sha384-sha256-ecp256-modp1024
   }
}

authorities {
   strongswan {
      cacert = strongswanCert.pem
      cert_uri_base = http://xxxx/
   }
}

As you can see from the following, I don't have any traffic selctors;

[ENC] generating IKE_AUTH request 1 [ IDi CERT N(HTTP_CERT_LOOK) CERTREQ AUTH CPRQ(ADDR DNS) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]

When I tested hash and url before, I did, but now it seems that they are not sent. Any ideas ?

ubuntu@ip-172-31-39-255:~$ sudo swanctl -i --ike home
[IKE] initiating IKE_SA home[40] to x.x.199.165
[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
[NET] sending packet: from 172.31.39.255[500] to x.x.199.165[500] (308 bytes)
[NET] received packet: from x.x.199.165[500] to 172.31.39.255[500] (359 bytes)
[ENC] parsed IKE_SA_INIT response 0 [ SA KE No V V V V N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(HTTP_CERT_LOOK) ]
[IKE] received Cisco Delete Reason vendor ID
[ENC] received unknown vendor ID: 43:49:53:43:4f:56:50:4e:2d:52:45:56:2d:30:32
[ENC] received unknown vendor ID: 43:49:53:43:4f:2d:44:59:4e:41:4d:49:43:2d:52:4f:55:54:45
[IKE] received Cisco FlexVPN Supported vendor ID
[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_384_192/PRF_HMAC_SHA2_384/ECP_256
[IKE] local host is behind NAT, sending keep alives
[IKE] remote host is behind NAT
[IKE] cert payload ANY not supported - ignored
[IKE] sending cert request for "CN=My VPN Server CA"
[IKE] sending cert request for "C=CH, O=strongSwan, CN=strongSwan Root CA"
[IKE] sending cert request for "C=CH, O=strongSwan, CN=strongSwan Root CA"
[IKE] sending cert request for "C=CH, O=strongSwan, CN=strongSwan Root CA"
[IKE] authentication of 'moon1.test.com' (myself) with RSA signature successful
[IKE] sending hash-and-url "http://x.x221/8ff08b63caaf4dc6e1e4c9feb419dc411a70d4a1"
[IKE] sending end entity cert "C=CH, O=strongswan, CN=moon.strongswan.org"
[ENC] generating IKE_AUTH request 1 [ IDi CERT N(HTTP_CERT_LOOK) CERTREQ AUTH CPRQ(ADDR DNS) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
[NET] sending packet: from 172.31.39.255[4500] to x.x.199.165[4500] (584 bytes)

**Comment by Thermi** [comment, marked as answer]
That is normal. They are sent as part of the a CREATE_CHILD_SA request.

**Reply by grahamqs** [reply]
Hi Thermi, thanks for this.

Looking at an old debug, it used to send this in IKE_AUTH;

[IKE] sending hash-and-url "[http://xxxxx/6fd9c99c3708d4ab502c65591d3f6a4adfa91478"](http://3.9.146.15/6fd9c99c3708d4ab502c65591d3f6a4adfa91478%22) 
[IKE] sending end entity cert "CN=vpn.example.com" 
[IKE] establishing CHILD_SA home{4} 
[ENC] generating IKE_AUTH request 1 [ IDi CERT N(INIT_CONTACT) N(HTTP_CERT_LOOK) CERTREQ IDr AUTH SA TSi TSr N(MOBIKE_SUP) N(NO_ADD_ADDR) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]

any idea what changed ?

**Reply by Thermi** [reply]
I see. It suspect it is depending on if you intend to create a CHILD_SA right away, or if it is a childlress IKE_SA.

You specified --ike, so it's a childless IKE_SA.

**Reply by grahamqs** [reply]
Ah ! I get you... 

Thanks, that worked;

$ sudo swanctl -i --ike home --child home

....

[IKE] sending end entity cert "C=CH, O=strongswan, CN=moon.strongswan.org"
[IKE] establishing CHILD_SA home{8}
[ENC] generating IKE_AUTH request 1 [ IDi CERT N(HTTP_CERT_LOOK) CERTREQ AUTH CPRQ(ADDR DNS) SA TSi TSr N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
